pkg/controller/vsphere/reconciler.go

```diff
@@ -389,6 +390,14 @@ func clone(s *machineScope) error {
 	var deviceSpecs []types.BaseVirtualDeviceConfigSpec
 	deviceSpecs = append(deviceSpecs, networkDevices...)
 
+	extraConfig := []types.BaseOptionValue{}
+
+	extraConfig = append(extraConfig, IgnitionConfig(userData)...)
```

Comment:
While it shouldn't happen because every machine will have user data, this might attempt to unpack a nil, which I think would cause the reconciler to crash.

Reply:
hmm.. based on https://play.golang.org/p/GVxF2yAVWKE
i don't this this should crash @michaelgugino  ?

Reply:
You are right, I tried it out myself.  Go is a hacky language.